modules/Bio/EnsEMBL/Variation/TranscriptVariationAllele.pm

```diff
@@ -729,12 +729,32 @@ sub hgvs_transcript {
   ### generic formatting 
   print "pre-format $hgvs_notation->{alt}\n" if $DEBUG ==1;
   $self->{hgvs_transcript} = format_hgvs_string( $hgvs_notation);
-
   if($DEBUG ==1){ print "HGVS notation: " . $self->{hgvs_transcript} . " \n"; }
 
+  ## save the HGVS style reference sequence in case the other long form of HGVS is required
+  $self->{hgvs_t_ref} = ($hgvs_notation->{type} eq 'dup') ? $hgvs_notation->{alt} :  $hgvs_notation->{ref};
+
   return $self->{hgvs_transcript}; 
 }
 
+=head2 hgvs_transcript_reference
+
+  Description: Return a string representing the reference sequence as could be used in HGVS notation
+               Useful for deletions where the recommended format formerly used the reference sequence
+               but no longer does.
+  Returntype : string or undef if no reference allele is available.
+  Exceptions : none
+  Status     : At Risk
+
+=cut
+
+sub hgvs_transcript_reference{
+
+  my $self = shift;
+  defined $self->{hgvs_t_ref} ? return $self->{hgvs_t_ref} : undef;
+
+}
+
```

Comment:
Could this method call `$self->hgvs_transcript` if `$self->{hgvs_t_ref}` is not defined, rather than returning `undef`? I think that would be neater, even if that avenue doesn't end up getting used.

Reply:
Yes - I was not sure how often hgvs_transcript would return nothing for a TVA and how slow it would be to call it twice. Do you reckon it's significant?

Reply:
You could set `$self->{hgvs_t_ref}` to undef at the beginning of `hgvs_transcript()`, then check `exists $self->{hgvs_t_ref}` rather than `defined`